web_fragments/tests/test_fragment.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Unit tests for the Fragment class.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+import ddt
+
+from django.test import TestCase
+
+from web_fragments.fragment import Fragment, FragmentResource
+from web_fragments.test_utils import (
+    CSS_EXPECTED_HTML, CSS_LINK_EXPECTED_HTML, EXPECTED_JS_INIT_VERSION, JS_EXPECTED_HTML,
+    JS_LINK_EXPECTED_HTML, TEST_CSS, TEST_CSS_URL, TEST_HTML, TEST_JS, TEST_JS_INIT_FN,
+    TEST_JS_URL, TEST_JSON_INIT_ARGS
+)
+
+
+@ddt.ddt
+class TestFragment(TestCase):
+    """
+    Unit tests for fragments.
+    """
+
+    def setUp(self):
+        super(TestFragment, self).setUp()
+
+    def create_test_fragment(self):
+        """
+        Creates a fragment for use in unit tests.
+        """
+        fragment = Fragment()
+        fragment.add_content(TEST_HTML)
+        fragment.add_css(TEST_CSS)
+        fragment.add_css_url(TEST_CSS_URL)
+        fragment.add_javascript(TEST_JS)
+        fragment.add_javascript_url(TEST_JS_URL)
+        fragment.initialize_js(TEST_JS_INIT_FN, json_args=TEST_JSON_INIT_ARGS)
+        return fragment
+
+    def validate_fragment(self, fragment=None, fragment_dict=None):
+        """
+        Validates that the fields of a fragment are all correct.
+        """
+        fragment_dict = fragment_dict if fragment_dict else fragment.to_dict()
+        self.assertEqual(fragment_dict['content'], TEST_HTML)
+        self.assertEqual(fragment_dict['js_init_fn'], TEST_JS_INIT_FN)
+        self.assertEqual(fragment_dict['js_init_version'], EXPECTED_JS_INIT_VERSION)
+        self.assertEqual(fragment_dict['json_init_args'], TEST_JSON_INIT_ARGS)
+        self.assertEqual(fragment_dict['resources'], self.test_resources_dict())
```

Comment:
I agree with this recommendation in general, but I think it is important that this particular contract be tested. Just determining that the text is somewhere in the dict doesn't seem to me to ensure that we won't break clients. Since this field is created by this library, it seems to me that it is reasonable to verify that it is exactly what is needed.